pkg/transport/fetchidentity.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package transport
+
+import (
+	"context"
+	"net"
+
+	"github.com/zeebo/errs"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/peertls"
+)
+
+// TODO: check whether this would be problematic with concurrency or should we thread the authinfo through context?
```

Comment:
I'm worried about this TODO. I don't want data races causing weird memory corruption because `AuthInfo` is an interface and so you can tear the writes. It's not too hard to just add a mutex.

Reply:
Yeah, it's less about concurrent writes, I'm not sure how grpc internally calls the handshakes, i.e. does it call it once, in parallel, does it cache or something else.

Reply:
Either way, I'll add a mutex.